### PR TITLE
Updating module for Terraform 0.13 and removing nested provider configs

### PR DESCRIPTION
**Pull request by danjbh**
## what
* Updating all modules to Terraform 0.13
* Removing nested 'opsgenie' provider configurations within each module
* Remove references to provider configuration in documentation
* Updating test cases

## why
* 0.13 allows us to iterate over modules with `for_each`, which feels like a dream come true
* Nested provider configurations do not work in 0.13 with `for_each` in modules, so we're ditching them

## references
* https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-13/

### CODE REVIEW

File: examples/api_integration/versions.tf [hunk 1]
```diff
@@ -1,7 +1,10 @@
 terraform {
-  required_version = "~> 0.12"
+  required_version = "~> 0.13"
```
Comment (osterman) [inline examples/api_integration/versions.tf:2]:
```suggestion
  required_version = "=> 0.12"
```

**Comment by 3h4x** [comment]
@danjbh Good stuff.

> Nested provider configurations do not work in 0.13 with for_each in modules, so we're ditching them

Didn't know that

**Pull request merged**